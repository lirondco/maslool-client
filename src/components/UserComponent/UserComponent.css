.UserComponent {
    margin-bottom: 2%;
    padding: 1%;
    border: #524c57 solid;
    border-radius: 10px;
    box-shadow: 5px 5px 2.5px grey;
  }

  .join_date {
      font-style: italic;
  }
 

  .FlaggedComment p {
      margin-left: 2%;
  }

  .UserComponent a {
      color: blue;
      text-decoration: underline;
  }

  .UserComponent a:hover {
    color: red;
    text-decoration: underline;
}


.banned_by {
    font-style: italic;
    color: red;
  }