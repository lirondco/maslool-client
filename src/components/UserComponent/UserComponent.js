import { Button, NiceDate } from '../Utils/Utils'
import React, { Component } from 'react'
import './UserComponent.css'

export default class UserComponent extends Component {
    static defaultProps = {
        onBanUser: () => {}
    }

    handleBanClick = (ev) => {
        ev.preventDefault()
        window.confirm('Are you sure you want to do this?')
        this.props.onBanUser()
    }

    render() {
        const user = this.props.user

        return (
            <div className='UserComponent'>
                <h4>{user.username}{user.banned && ' [banned]'}{user.admin && ' [moderator]'}</h4>
                <p className='join_date'>Joined <NiceDate date={user.join_date} /></p>
                <p><a href={`mailto:${user.email}`}>{user.email}</a></p>
                {user.banned && <p className='banned_by'>Banned by: {user.banned_by}</p>}
                {user.banned ? <Button type='button' onClick={this.handleBanClick}>Unban User</Button> : <Button disabled={user.admin} type='button' onClick={this.handleBanClick}>Ban User</Button>}
            </div>
        )
    }
} 