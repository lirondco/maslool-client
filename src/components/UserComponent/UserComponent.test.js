import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import UserComponent from './UserComponent'

describe(`UserComponent`, () => {
    const props = {
        id: 1,
        email: 'test@ema.il',
        username: 'testuser',
        join_date: '2021-03-11T06:47:06.477Z',
        admin: false,
        banned: false,
        banned_by: null
    }

    it('renders the component given props', () => {
        const wrapper = shallow(<UserComponent user={props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('button works when clicked', () => {
        const wrapper = shallow(<UserComponent user={props} />)
        window.confirm = jest.fn(() => true)
        wrapper.find('ForwardRef').simulate('click', { preventDefault() {}})
        expect(window.confirm).toBeCalled()
    })
})