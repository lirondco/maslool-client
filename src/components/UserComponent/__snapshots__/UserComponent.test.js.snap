// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserComponent renders the component given props 1`] = `
<div
  className="UserComponent"
>
  <h4>
    testuser
  </h4>
  <p
    className="join_date"
  >
    Joined 
    <NiceDate
      date="2021-03-11T06:47:06.477Z"
    />
  </p>
  <p>
    <a
      href="mailto:test@ema.il"
    >
      test@ema.il
    </a>
  </p>
  <ForwardRef
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Ban User
  </ForwardRef>
</div>
`;
